src/flb_pack.c

```diff
@@ -923,13 +924,23 @@ flb_sds_t flb_pack_msgpack_to_json_format(const char *data, uint64_t bytes,
             case FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP:
             /* Format the time, use microsecond precision not nanoseconds */
                 gmtime_r(&tms.tm.tv_sec, &tm);
-                s = strftime(time_formatted, sizeof(time_formatted) - 1,
+                s = strftime(time_formatted, sizeof(time_formatted),
                              FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP_FMT, &tm);
 
+                max_len = sizeof(time_formatted) - s;
                 len = snprintf(time_formatted + s,
-                               sizeof(time_formatted) - 1 - s,
+                               max_len,
                                ".%06" PRIu64,
                                (uint64_t) tms.tm.tv_nsec / 1000);
+                if (len >= max_len){
+                    if (json_format == FLB_PACK_JSON_FORMAT_LINES ||
+                        json_format == FLB_PACK_JSON_FORMAT_STREAM) {
+                            flb_sds_destroy(out_buf);
+                        }
```

Comment:
Could you check indentation level ?

Reply:
I thought it will be more readable. Anyway done.

Reply:
Oops, I meant like below.
Could you fix it ?

```c
                    if (json_format == FLB_PACK_JSON_FORMAT_LINES ||
                        json_format == FLB_PACK_JSON_FORMAT_STREAM) {
                        flb_sds_destroy(out_buf);
                    }
                    msgpack_sbuffer_destroy(&tmp_sbuf);
```

Reply:
Now I see it! Done.